### ISSUE TITLE
Cannot install on distros where package repos don't include GLIBC 2.38

### ISSUE THREAD

**Issue by clement-or** [zellij-org/zellij#3085]
## Issue description
On distros where repos don't include the latest GLIBC version (such as debian-based distros), zellij cannot be used. Please provide a way to either embed GLIBC in the zellij install, or compile zellij against another GLIBC version.

Ideally, I'd be a strong advocate for not relying on C libs and make zellij a fully rust native app, but I have no idea what changes it would implicate.

## Minimal reproduction
* Install zellij on Ubuntu 22.04

## Other relevant information
/etc/os-release
```
NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os
```

**Comment by imsnif** [comment]
Could you please provide more information? I'm not familiar with this issue (and don't have access to an Ubuntu machine).

Also, as an aside: you can always use the precompiled musl versions.

**Comment by clement-or** [comment]
Here are reproduction steps : 

* Use Ubuntu 22.04
* Install zellij using cargo `cargo install zellij`
* Start zellij with `zellij`

Output: 
```
zellij: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found (required by zellij)
zellij: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by zellij)
```

**Comment by imsnif** [comment]
While I don't know a lot about Ubuntu, these errors seem to me like some configuration issue on the machine. Since I don't have access to Ubuntu, I cannot confirm this myself.

I guess if this is a systematic problem beyond this specific machine we'll get a lot more people reporting this. So let's wait and see.

Meanwhile, as I mentioned, you can use the precompiled musl version from the releases. Thanks for reporting this, in any case.

**Comment by Zykino** [comment]
All rust binaries depends on libc. That’s how the language is made…

I’m using zellij in WSL on both Debian 10 (libc 2.28) and 12 (libc 2.36) without any issue. Even on my pop-os (libc 2.34) (which have the exact same `/etc/os-release` file as you) zellij work without any issue.

If you do `dpkg -l "*libc*"` is libc installed?
There should be one line, starting with ii like "ii  libc6:amd64" (amd64 is here for my x86 64bits system, might be a different one depending on the hardware you execute on)

**Comment by Zykino** [comment]
Also note that zellij is exclusively tested with the locked libs. Install it with the recommended command: `cargo install --locked zellij`.
It may happen for any rust binary that fail installing/executing: first thing to try is installing with the `--locked` option.

**Comment by guihkx** [comment]
I don't think this makes any sense, because Ubuntu 22.04 only has glibc 2.35, so for the `zellij` binary to require symbols from glibc 3.38, that version would have to be installed on the machine you built it in the first place - as far as I know, anyway.

> Here are reproduction steps :
> 
>     * Use Ubuntu 22.04
> 
>     * Install zellij using cargo `cargo install zellij`
> 
>     * Start zellij with `zellij`
> 
> 
> Output:
> 
> ```
> zellij: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found (required by zellij)
> zellij: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by zellij)
> ```

This also does not reproduce for me on a clean image of Ubuntu 22.04 running on Docker.

@clement-or Are you sure you're actually invoking the `zellij` binary built by cargo? The install path defaults to `~/.cargo/bin`, which should also be in your `PATH`, i.e.:

```
$ which zellij
/home/me/.cargo/bin/zellij
```
```
$ zellij --version
zellij 0.39.2
```
```
$ ldd --version
ldd (Ubuntu GLIBC 2.35-0ubuntu3.6) 2.35
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
```
```
$ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

**Comment by clement-or** [comment]
Did a bit more troubleshooting, I guess it's something with my install, but zellij is the only cargo package I've had trouble with.

First, I ran `dpkg -l "*libc*"` as suggested and libc6:amd64 is installed (I run a regular x84_64 machine). The version is 2.35-0ubuntu3.6 (latest provided by the repos)

On trying `ldd --version`, I notice LDD is in version 2.38. It may be a result of my attempts to manually install the latest version of glibc before opening this issue. 

Few weeks have passed, I tried reinstalling zellij with the --locked flag, which yields a different log. In fact zellij no longer installs. I probably have changed things on my machine (probably updated packages). I know this is not helpful, I'd like to help find a repro if possible.

Now I just get
```bash
$ cargo install --locked zellij
;; Snip
error: linking with `cc` failed: exit status: 1
```
